lib/CXGN/Trial/TrialDesign.pm

```diff
@@ -463,13 +633,54 @@ sub _get_alpha_lattice_design {
   @rep_numbers = $result_matrix->get_column("replication");
   @stock_names = $result_matrix->get_column("trt");
   @converted_plot_numbers=@{_convert_plot_numbers($self,\@plot_numbers)};
+
+  if ($plot_layout_format eq "zigzag") {
+    if (!$fieldmap_col_number){
+      @col_number_fieldmaps = ((1..$number_of_blocks) x ($number_of_blocks * $number_of_reps));
+      print STDERR Dumper(\@col_number_fieldmaps);
+    } else {
+        @col_number_fieldmaps = ((1..$fieldmap_col_number) x $fieldmap_row_number);
+      }
+  }
+  elsif ($plot_layout_format eq "serpentine") {
+    if (!$fieldmap_row_number)  {
+      for my $rep (1 .. ($number_of_blocks * $number_of_reps)){
+        if ($rep % 2){
+          push @col_number_fieldmaps, (1..$number_of_blocks);
+        } else {
+          push @col_number_fieldmaps, (reverse 1..$number_of_blocks);
+        }
+      }
+    } else {
+        for my $rep (1 .. $fieldmap_row_number){
+          if ($rep % 2){
+            push @col_number_fieldmaps, (1..$fieldmap_col_number);
+          } else {
+              push @col_number_fieldmaps, (reverse 1..$fieldmap_col_number);
+          }
+        }
+      }
+  }
+
+  if ($plot_layout_format && !$fieldmap_col_number && !$fieldmap_row_number){
+    @fieldmap_row_numbers = (@block_numbers);
+  }
+  elsif ($plot_layout_format && $fieldmap_row_number){
+      @fieldmap_row_numbers = ((1..$fieldmap_row_number) x $fieldmap_col_number);
+      @fieldmap_row_numbers = sort {$a <=> $b} @fieldmap_row_numbers;
+    }
+
   for (my $i = 0; $i < scalar(@converted_plot_numbers); $i++) {
     my %plot_info;
     $plot_info{'stock_name'} = $stock_names[$i];
     $plot_info{'block_number'} = $block_numbers[$i];
     $plot_info{'plot_name'} = $converted_plot_numbers[$i];
     $plot_info{'rep_number'} = $rep_numbers[$i];
     $plot_info{'is_a_control'} = exists($control_names_lookup{$stock_names[$i]});
+    if ($fieldmap_row_numbers[$i]){
```

Comment:
for alpha design check page 20 for enumeration of plots. https://cran.r-project.org/web/packages/agricolae/vignettes/tutorial.pdf